pilot/cmd/pilot-discovery/main.go

```diff
@@ -126,6 +126,8 @@ func init() {
 		"Select a namespace where the controller resides. If not set, uses ${POD_NAMESPACE} environment variable")
 	discoveryCmd.PersistentFlags().StringSliceVar(&serverArgs.Plugins, "plugins", bootstrap.DefaultPlugins,
 		"comma separated list of networking plugins to enable")
+	discoveryCmd.PersistentFlags().BoolVar(&serverArgs.InjectPodRedirectAnnot, "injectPodRedirectAnnot", false,
```

Comment:
I think as this will be a 2.0 release, we can just remove this option altogether, always enabling it. That doesn't have to happen in this PR though. I filed a ticket to do it: https://issues.redhat.com/browse/MAISTRA-1612